Add dynamic model repository selection for Whisper based on environment variable

stisiTT opened the pull request:
Single card tests [PASS](https://github.com/tenstorrent/tt-metal/actions/runs/16787153389/job/47540978650)
Blackhole demo tests [PASS](https://github.com/tenstorrent/tt-metal/actions/runs/16809281134)

Add dynamic model repository selection for Whisper based on environment variable

- Introduced `get_whisper_model_repo` function to determine the model repository based on the `WHISPER_MODEL_TYPE` environment variable, supporting both 'original' and 'distilled' models.
- Updated model loading functions to utilize the new repository selection logic, enhancing flexibility for model deployment.
- Adjusted comments for clarity regarding model length configurations.

stisiTT committed:
Updating weights for non-distilled Whisper model

stisiTT committed:
Add dynamic model repository selection for Whisper based on environment variable

- Introduced `get_whisper_model_repo` function to determine the model repository based on the `WHISPER_MODEL_TYPE` environment variable, supporting both 'original' and 'distilled' models.
- Updated model loading functions to utilize the new repository selection logic, enhancing flexibility for model deployment.
- Adjusted comments for clarity regarding model length configurations.

stisiTT committed:
Pre-commit Format update

stisiTT commented on models/demos/whisper/demo/demo.py:
```diff
@@ -49,12 +49,22 @@ def pad_input_32(tensor, value):
     return tensor
 
 
+def get_whisper_model_repo():
+    """Get whisper model repo based on environment variable"""
+    model_type = os.environ.get("WHISPER_MODEL_TYPE", "original")
+    if model_type == "distilled":
+        return "distil-whisper/distil-large-v3"
+    elif model_type == "original":
+        return "openai/whisper-large-v3"
+    else:
+        raise ValueError(f"Unknown WHISPER_MODEL_TYPE: {model_type}. Valid options are 'original' or 'distilled'")
```
Happy to!

bgoelTT reviewed (commented):
@stisiTT can you modify this function https://github.com/tenstorrent/tt-metal/blob/221891bae569c56c8645da93ddd335f1e0ad3f95/models/demos/whisper/demo/demo.py#L234, to accept the hugging face repo id as an argument and pass it down to loading the HF model?

If provided, the demo tests should now pass down the user-specified hugging face repo id into this function. This will allow us to just update `create_functional_whisper_for_conditional_generation_inference_pipeline()` in the eval code and eventually Whisper-X

stisiTT committed:
PyTest Function update

skhorasganiTT commented on models/demos/whisper/demo/demo.py:
```diff
@@ -49,12 +49,23 @@ def pad_input_32(tensor, value):
-    )
-    processor = AutoProcessor.from_pretrained("distil-whisper/distil-large-v3", language="English", task="transcribe")
-    feature_extractor = AutoFeatureExtractor.from_pretrained("distil-whisper/distil-large-v3")
+def get_whisper_model_repo(model_type="original"):
+    """
+    Get whisper model repo based on model type parameter
+    """
+    if model_type == "distilled":
+        return "distil-whisper/distil-large-v3"
+    elif model_type == "original":
+        return "openai/whisper-large-v3"
+    else:
+        raise ValueError(f"Unknown model_type: {model_type}. Valid options are 'original' or 'distilled'")
+
+
+def load_conditional_generation_ref_model(model_type="original"):
+    model_repo = get_whisper_model_repo(model_type)
```
Could you remove `model_type` /  `get_whisper_model_repo` and just pass in the hf name to the test (and check in `load_conditional_generation_ref_model` whether the name is in ["distil-whisper/distil-large-v3", "openai/whisper-large-v3"]?

skhorasganiTT commented on models/demos/whisper/demo/demo.py:
```diff
@@ -465,10 +482,14 @@ def test_demo_for_audio_classification_dataset(ttnn_model, device, is_ci_env):
+def test_demo_for_conditional_generation(input_path, ttnn_model, device, num_inputs, model_type, is_ci_env):
     ttft, decode_throughput = run_demo_whisper_for_conditional_generation_inference(
-        input_path, ttnn_model, device, num_inputs
+        input_path, ttnn_model, device, num_inputs, model_type
     )
     if is_ci_env:
```
Please modify the `if is_ci_env` check to also check that the model type is the distilled version so we only assert on perf for that model. 
You can run single card demo tests to verify the test passes - https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml (Note: you only care about the whisper-n300-perf job in that pipeline).

stisiTT committed:
Refactor Whisper model loading functions to use a single model repository parameter

- Removed the `get_whisper_model_repo` function and replaced it with a direct model repository check in `load_conditional_generation_ref_model`.
- Updated function signatures and internal calls to use `model_repo` instead of `model_type`, simplifying the model selection process.
- Adjusted tests to reflect the new parameterization for model repository, ensuring compatibility with both 'original' and 'distilled' models.

stisiTT committed:
Update CI environment check for Whisper demo to include specific model repository condition

- Modified the conditional check in `test_demo_for_conditional_generation` to ensure performance metrics are only evaluated when the environment is CI and the model repository is set to "distil-whisper/distil-large-v3". This enhances the accuracy of performance testing in CI environments.

skhorasganiTT commented on models/demos/whisper/demo/demo.py:
```diff
@@ -231,14 +240,19 @@ def _run_generate():
     the callable returns the full decoded output.
+    
+    Args:
+        ttnn_model: The compiled TTNN model
```
You can remove "compiled" as it's not compiled until it's run the first time

skhorasganiTT commented on models/demos/whisper/demo/demo.py:
```diff
@@ -231,14 +240,19 @@ def _run_generate():
 
 
-def create_functional_whisper_for_conditional_generation_inference_pipeline(ttnn_model, device):
+def create_functional_whisper_for_conditional_generation_inference_pipeline(ttnn_model, device, model_repo="openai/whisper-large-v3"):
```
Would prefer if default value for model_repo is not set (in all the functions) to ensure it's always passed in as desired, but not big deal.

skhorasganiTT reviewed (approved):
Thanks @stisiTT! Please ensure that the single card demo tests pass (and post the pipeline link in the PR description) before attempting to merge.

stisiTT committed:
rebase and pre-commit updates

stisiTT committed:
Small changes to language and defaults in functions

skhorasganiTT commented:
> @skhorasganiTT [here](https://github.com/tenstorrent/tt-metal/actions/runs/16787153389/job/47540978650), you can see that the whisper test passed, but the overall test failed due to models not associated with whisper. Would that still be on me? What's best practice, and what is acceptable in this case?

Thanks Sam! No you don't need to worry about the other failing models, just Whisper. 
I forgot to mention before but could you please also run the blackhole demo tests & verify the Whisper jobs pass (https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-demo-tests.yaml)? If that passes as well you can feel free to merge.
